cmd/snapd-apparmor/snapd-apparmor

```diff
@@ -49,6 +49,11 @@ is_container_with_internal_policy() {
 	# shellcheck disable=SC3043,SC2039
 	local ns_name
 
+	# WSL needs to be detected explicitly
+	if [ $(systemd-detect-virt --container) = "wsl" ]; then
+		return 0
```

Comment:
Given that WSL runs one kernel and many distributions on top, will snapd's usage of apparmor in one such distribution leak to other distributions on the same system?

Reply:
At the moment, the answer to that is... it depends? There's a discussion I had over at the AppArmor version of this patch touching on that point which might be worth looking at, but to summarize:

It depends on how people set up their systems. snapd requires systemd, so at the moment, snapd on WSL requires using one of the tools people have built to run systemd on WSL. I've modified mine ([genie](https://github.com/arkane-systems/genie)) in the upcoming release 1.45 to explicitly create an AppArmor policy namespace to prevent leakage of AppArmor policies, and will drop a quick note to the authors of the other three I'm aware of suggesting they do the same. This should eliminate the likelihood of this happening in practice.

On the other hand, if you roll your own systemd hack and _don't_ do this, things could leak, yes.

(On the gripping hand, since the wall between distributions in WSL is just a (pid, mnt) namespace and nothing more, anyone who uses multiple distros under WSL should already be aware of and used to be various things leaking and doing the needful, I should expect.)